### ISSUE TITLE
Update to `versioned` accessibility-checker config, latest ruleset

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#17149]
In https://github.com/carbon-design-system/carbon/pull/17128 accessibility-checker was updated. We can now make use of the `versioned` ruleset.

`versioned` makes sure we use a rule pack that matches the version of accessibility-checker we're currently on. New rulesets will only be brought in by updating accessibility-checker, so no more randomly failing CI checks out of the blue after a new ruleset is published on latest. 

This will effectively update the ruleset to the latest, `06Aug2024`. There will be violations/failures we'll need to sort through and fix and/or turn off certain rules that do not apply to storybook or the context in which we're developing the components.

```[tasklist]
### Potential related violations
- [ ] https://github.com/carbon-design-system/carbon/issues/17176
- [ ] https://github.com/carbon-design-system/carbon/issues/17223
- [ ] From #16582: 1. DatePicker @avt-advanced-states open test skipped due to color contrast a11y failure and 2. Skipped DataTable's @avt-advanced-states with-radio-expansion test due to an odd failure where the id's on rows are not correct on the dynamic story
- [ ] https://github.com/carbon-design-system/carbon/issues/18202
```
